core/src/main/scala/cromwell/util/TryWithResource.scala

```diff
@@ -0,0 +1,41 @@
+package cromwell.util
+
+import scala.util.Try
+
+/**
+  * If this looks java-y, it's because it is. It's (almost) a direct-to-scala mapping of the Java try-with-resources block.
+  * This code looks ugly so yours won't have to! (TM)
+  */
+class TryWithResource[A <: AutoCloseable](getResource: () => A) {
+  def into[B](block: A => B): Try[B] = Try {
```

Comment:
Though I guess it's called only from the companion object so maybe that doesn't matter.  

If you want to restrict this to being called by the companion object maybe it should be private?